File: bug65_single_element_case_failure/bug65_single_element_case_failure/main1.java
public class main1 {
    public static void main(String[] args) {
        int[] data = {0xABCD};
        int pairs_len = data.length - 1;

        // ❌ REMOVED: wrong error condition
        // if (pairs_len <= 0) {
        //     System.exit(1);
        // }

        int[] pairs = new int[pairs_len];

        for (int i = 0; i < pairs_len; i++) {
            pairs[i] = data[i] ^ data[i+1];
        }

        long acc = 0;

        for (int i = 0; i < pairs_len; i++) {
            acc ^= pairs[i] * (i + 1);
            acc = (acc * 0x9E3779B97F4A7C15L);
        }

        System.out.println(Long.toHexString(acc)); // ✅ Added output
    }
}

File: bug65_single_element_case_failure/bug65_single_element_case_failure/main1.js
function main() {
    let data = [0xABCD];
    let pairs = [];

    for (let i = 0; i < data.length - 1; i++) {
        pairs.push(data[i] ^ data[i + 1]);
    }

    // ❌ REMOVED: wrong error handling
    // if (pairs.length === 0) {
    //     process.exit(1);
    // }

    let acc = 0n;

    for (let i = 0; i < pairs.length; i++) {
        acc ^= BigInt(pairs[i]) * BigInt(i + 1);
        acc = (acc * 0x9E3779B97F4A7C15n) & 0xFFFFFFFFFFFFFFFFn;
    }

    console.log(acc.toString(16)); // ✅ Added output
}
main();